library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -577,12 +584,32 @@ public synchronized void onSetUserCompleted(ClientConnectionCallback callback) {
 
     private synchronized void connect() {
         Log.i(TAG, "client.connect was called");
+
+        if (anonymousConnection) {
```

Comment:
perhaps avoid kill code duplication:

- remove `anonymousConnect` method
- revert `anonymousConnect` to be called `connect`
- inside `connect` pass `anonymousConnection` as `webSocketServiceProvider.provideWebSocketService`'s last arg  (same thing for `provideApiService`)